blockstack/blockstack-browser
# Remove dependency on local core endpoint

This PR removes `blockstack-browser`'s dependency on the local core endpoint and enables the bitcoin wallet and name registration processes in the Windows and the webapp builds. These are now identical to the normal build, except that `isCoreEndpointDisabled` will _always_ return `true` on those builds-- on the normal build, when using the regtest environment, that returns `false`.

This PR incorporates changes from the `feature/client-side-btc` #1254 (that needs to be approved before this should be), but not #1255.

# Testing

### Regtest Testing

To test in regtest, start the docker container:

```
$ IMAGE=$(docker run -dt -p 6270:6270 -p 16269:16269 -p 18332:18332 -e BLOCKSTACK_TEST_CLIENT_RPC_PORT
=6270 -e BLOCKSTACK_TEST_CLIENT_BIND=0.0.0.0 -e BLOCKSTACK_TEST_BITCOIND_ALLOWIP=172.17.0.0/16 quay.io/blockstack/inte
grationtests:master blockstack-test-scenario --interactive 2 blockstack_integration_tests.scenarios.browser_env)
$ docker logs -f $IMAGE | grep -q 'Test finished'
```

Once that's initialized, you can run the browser proxy with:

```
$ npm i && npm run dev-proxy & npm run dev
```

If you open `localhost:3000/#coreAPIPassword=blockstack_integration_test_api_password&regtest=1`, it will activate regtestmode, and you should be able to register names, send bitcoin transactions, etc. (once you've loaded up the browser's client-side wallet from the `wallet/send-core` page).

### Mainnet testing

Run the browser, with the dev-proxy, and open `localhost:3000`. You _should_ reset your browser's localStorage, because this build changes around the default API endpoints.

Now -- you should be able to log into apps, just as before, though applications that use deprecated storage will no longer work -- try coinkitty:

![old-gaia](https://user-images.githubusercontent.com/1153552/37726958-04e3b840-2d0d-11e8-9e39-14d947f07892.png)
